### PR TITLE
Homepage now rendered 100% static

### PR DESCRIPTION
**Pull request by patricksimonian**
## Summary
Converted the homepage to render statically

## Notable Changes 
- integrate React Hooks for searching
- greatly improved homepage unit test spec
- improved SEO by added meta description tag
- insane boost in performance!

/bot-ignore-length

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (ShellyXueHan) [inline .vscode/launch.json]: Do you want to keep this?

File: app-web/__tests__/pages/Index.test.js [hunk 2]
```diff
@@ -1,116 +1,142 @@
...
+  test('when searching, collections disappear if there are results', () => {
+    queryString.parse.mockReturnValue({ q: 'foo' });
+    useSearch.mockReturnValue([]);
+    const { queryByTestId } = render(
+      <ThemeProvider theme={theme}>
+        <Index {...props} />
+      </ThemeProvider>,
+    );
+
+    expect(queryByTestId('collections-container')).not.toBeInTheDocument();
```
Comment (ShellyXueHan) [inline app-web/__tests__/pages/Index.test.js]: and assert that results show?

File: app-web/__tests__/pages/Index.test.js [hunk 3]
```diff
@@ -1,116 +1,142 @@
...
 
-    test('when search props updates, it calls search function', async () => {
-      const wrapper = shallow(<Index {...props} />);
-      wrapper.setProps({ location: { ...location, search: '?q=foo' } });
-      await wrapper.instance().componentDidMount();
-      expect(getSearchResults).toHaveBeenCalled();
-      expect(props.setSearchQuery).toHaveBeenCalled();
-      expect(props.setSearchResults).toHaveBeenCalled();
-    });
+    expect(getByTestId(COLLECTION_TEST_IDS.container)).toBeInTheDocument();
```
Comment (ShellyXueHan) [inline app-web/__tests__/pages/Index.test.js]: If `getByTestId` throws, reconsider on asserting it.

File: app-web/src/pages/index.js [hunk 4]
```diff
@@ -1,181 +1,160 @@
...
+  > p {
+    margin-bottom: 20px;
+    max-width: 738px;
+    font-size: 16px;
+    font-weight: 400;
+    line-height: 1.3em;
+  }
+`;
+
+const collectionsSelector = selectCollectionsWithResourcesGroupedByType();
```
Comment (ShellyXueHan) [inline app-web/src/pages/index.js]: moving them

File: app-web/__tests__/pages/Index.test.js (hunk 2 above)
Comment (patricksimonian) [reply app-web/__tests__/pages/Index.test.js]: done

File: app-web/__tests__/pages/Index.test.js (hunk 3 above)
Comment (patricksimonian) [reply app-web/__tests__/pages/Index.test.js]: changed to query by test id which does not throw

**Pull request merged**